### DISCUSSION TITLE
Intuition for complex model topology?

### DISCUSSION THREAD

**Discussion by mvenouziou** [ultralytics/yolov5#4091] [Q&A]
Does anyone have an intuitive explanation for the Yolo v5 model graph? I feel I understand v1-v4 backbone/head pretty well, but I can't imagine how people came up with v5. It clearly works very well, I just have no idea why!

https://user-images.githubusercontent.com/26833433/114286928-349bd600-9a63-11eb-941f-7139ee6cd602.png

**Comment by glenn-jocher** [comment]
@mvenouziou the architecture is based on many thousands of man hours of R&D and countless cups of coffee.

**Reply by mvenouziou** [reply]
I believe it! It took me a lot of effort and more than just a few peeks at the official class codes, topology diagram and v1-v4 papers in just to re-code it in Tensorflow. I still can't believe how fast it runs on CPU, that's quite an achievement!
